src/extensions/extensionInterface.ts

```diff
@@ -289,7 +295,14 @@ export const recordingSelectionReducer: Reducer<RecordingSelection, RecordingSel
     else if (action.type === 'SetSelectedElectrodeIds') {
         return {
             ...state,
-            selectedElectrodeIds: action.selectedElectrodeIds
+            selectedElectrodeIds: action.selectedElectrodeIds.filter(eid => ((!state.visibleElectrodeIds) || (state.visibleElectrodeIds.includes(eid))))
+        }
+    }
+    else if (action.type === 'SetVisibleElectrodeIds') {
+        return {
+            ...state,
+            visibleElectrodeIds: action.visibleElectrodeIds,
+            selectedElectrodeIds: state.selectedElectrodeIds ? state.selectedElectrodeIds.filter(eid => (action.visibleElectrodeIds.includes(eid))) : undefined
```

Comment:
Enforce that selected electrode IDs must be among the visible IDs

Reply:
And to clear a subsetting restriction, we call this verb with a `null` value for `action.visibleElectrodeIds`.

Reply:
yes